src/modules/users/users.repository.ts

```diff
@@ -16,19 +20,72 @@ export class UsersRepository {
         email,
         password,
         displayName: `${name} ${lastName}`,
+        refreshToken: '',
+      },
+    });
+  }
+
+  async createWithRole(createUserDto: CreateUserDto, role: Role) {
```

Comment:
I realized that a specific user could have more than one role, is this the right approach? Instead of two different methods to create a user, it could be better to simplify all this logic in a single function.